pkg/i2gw/providers/istio/istio.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package istio
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// The ProviderName returned to the provider's registry.
+const ProviderName = "istio"
+
+func init() {
+	i2gw.ProviderConstructorByName[ProviderName] = NewProvider
+}
+
+type Provider struct {
+	storage   storage
+	reader    reader
+	converter converter
+}
+
+// NewProvider returns the istio implementation of i2gw.Provider.
+func NewProvider(config *i2gw.ProviderConf) i2gw.Provider {
```

Comment:
```suggestion
func NewProvider(conf *i2gw.ProviderConf) i2gw.Provider {
```

nit: for consistency with other implementations

Reply:
done